lib/node_modules/@stdlib/stats/base/dists/cosine/mgf/test/test.native.js

```diff
@@ -0,0 +1,98 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var resolve = require( 'path' ).resolve;
+var tape = require( 'tape' );
+var isNumber = require( '@stdlib/assert/is-number' ).isPrimitive;
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var randu = require( '@stdlib/random/base/randu' );
+var PINF = require( '@stdlib/constants/float64/pinf' );
+var NINF = require( '@stdlib/constants/float64/ninf' );
+var tryRequire = require( '@stdlib/utils/try-require' );
+
+
+// VARIABLES //
+
+var mgf = tryRequire( resolve( __dirname, './../lib/native.js' ) );
+var opts = {
+	'skip': ( mgf instanceof Error )
+};
+
+
+// TESTS //
+
+tape( 'main export is a function', opts, function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof mgf, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'if provided `NaN` for any parameter, the function returns `NaN`', opts, function test( t ) {
+	var y = mgf( NaN, 0.0, 1.0 );
+	t.equal( isnan( y ), true, 'returns NaN' );
+	y = mgf( 0.0, NaN, 1.0 );
+	t.equal( isnan( y ), true, 'returns NaN' );
+	y = mgf( 0.0, 1.0, NaN );
+	t.equal( isnan( y ), true, 'returns NaN' );
+	t.end();
+});
+
+tape( 'if provided a negative `s`, the function returns `NaN`', opts, function test( t ) {
+	var y;
+
+	y = mgf( 2.0, 2.0, -1.0 );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	y = mgf( 0.0, 2.0, -1.0 );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	y = mgf( 2.0, 1.0, NINF );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	y = mgf( 2.0, PINF, NINF );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	y = mgf( 2.0, NINF, NINF );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	y = mgf( 2.0, NaN, NINF );
+	t.equal( isnan( y ), true, 'returns NaN' );
+
+	t.end();
+});
+
+tape( 'the function evaluates the MGF', opts, function test( t ) {
+	var mu;
+	var s;
+	var x;
+	var y;
+	var i;
+
+	for ( i = 0; i < 100; i++ ) {
+		x = randu() * 10.0;
+		mu = randu() * 10.0;
+		s = randu() * 10.0;
+		y = mgf( x, mu, s );
+		t.strictEqual( isNumber( y ), true, 'returns true' );
```

Comment:
We should check whether the cosine MGF function is nowadays available in either Julia, R, or Python. Would be good to have test fixtures for this package.